### ISSUE TITLE
CA2000 false positive in VS 17.8.0 Preview 2

### ISSUE THREAD

**Issue by jeremy-visionaid** [dotnet/roslyn-analyzers#6954]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/en-nz/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

VS 17.8.0 Preview 2

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100-rc.1.23455.8](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

CA2000 triggers when returning an object from a named constructor

### Steps To Reproduce

Minimal repro with SkiaSharp:

```
    private static SKBitmap CreateBitmap(SKSizeI size)
    {
        SKBitmap skBitmap = new(size.Width, size.Height);
        using SKCanvas canvas = new(skBitmap);
        Draw(canvas);
        return skBitmap;
    }

    private static void Draw(SKCanvas canvas)
    {
        using SKPaint paint = new();
        using SKPath path = new();
        path.LineTo(1, 1);
        path.Close();
        canvas.DrawPath(path, paint);
    }
```

### Expected behavior

No warning

### Actual behavior

CA2000 triggers for skBitmap

**Comment by RobSiklos** [comment]
@mavasani Not sure if it's exactly the same issue, but here's a repro which doesn't have any external dependencies:

```csharp
internal void CA2000Test()
{
    foreach (int i in new[] { 0, 1, 2 })
    {
        using (IDisposable d = new D())
        {
            Blah(d);
        }
    }
}

private void Blah(IDisposable d)
{
    using (var fs = new FileStream("abc", FileMode.Open)) { }
}

class D : IDisposable
{
    public void Dispose() { }
}
```

Notes:
* if you replace the `foreach` loop with a `for` loop, the warning goes away
* if you replace the `FileStream` with a `MemoryStream`, or any other type of `IDisposable` that I could find, the warning goes away.
